tests/operator_test.go

```diff
@@ -234,8 +287,107 @@ var _ = Describe("Operator delete CDI tests", func() {
 	})
+
+	It("Should install CDI infrastructure pods with node placement", func() {
+		By("Deleting CDI CR")
+		err := f.CdiClient.CdiV1beta1().CDIs().Delete(context.TODO(), cr.Name, metav1.DeleteOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Waiting for CDI CR and infra deployments to be deleted after CDI CR was removed")
+		Eventually(func() bool { return infraDeploymentGone(f) && crGone(f, cr) }, 15*time.Minute, 2*time.Second).Should(BeTrue())
+
+		By("Deleting CDI operator")
+		err = f.K8sClient.AppsV1().Deployments(f.CdiInstallNs).Delete(context.TODO(), "cdi-operator", metav1.DeleteOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Waiting for CDI operator to be deleted")
+		Eventually(func() bool { return cdiOperatorGone(f) }, 5*time.Minute, 2*time.Second).Should(BeTrue())
+
+		By("Creating modified CDI CR, with infra nodePlacement")
+		localSpec := cr.Spec.DeepCopy()
+		localSpec.Infra = tests.TestNodePlacementValues(f)
+
+		cdi := &cdiv1.CDI{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "cdi",
+			},
+			Spec: *localSpec,
+		}
+
+		_, err = f.CdiClient.CdiV1beta1().CDIs().Create(context.TODO(), cdi, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Recreating CDI operator")
+		_, err = f.K8sClient.AppsV1().Deployments(f.CdiInstallNs).Create(context.TODO(), cdiOperator, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Verifying CDI apiserver, deployment, uploadproxy exist, before continuing")
+		Eventually(func() bool { return infraDeploymentAvailable(f, cr) }, CompletionTimeout, assertionPollInterval).Should(BeTrue(), "Timeout reading CDI deployments")
+
+		By("Testing all infra deployments have the chosen node placement")
+		for _, deploymentName := range []string{"cdi-apiserver", "cdi-deployment", "cdi-uploadproxy"} {
+			deployment, err := f.K8sClient.AppsV1().Deployments(f.CdiInstallNs).Get(context.TODO(), deploymentName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			match := tests.PodSpecHasTestNodePlacementValues(f, deployment.Spec.Template.Spec)
+			Expect(match).To(BeTrue(), fmt.Sprintf("node placement in pod spec\n%v\n differs from node placement values in CDI CR\n%v\n", deployment.Spec.Template.Spec, cr.Spec.Workloads))
+		}
+
+		By("Deleting modified CDI operator and CDI CR, original will be re-created")
+		err = f.CdiClient.CdiV1beta1().CDIs().Delete(context.TODO(), cr.Name, metav1.DeleteOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		Eventually(func() bool { return infraDeploymentGone(f) && crGone(f, cr) }, 5*time.Minute, 2*time.Second).Should(BeTrue())
+
+		err = f.K8sClient.AppsV1().Deployments(f.CdiInstallNs).Delete(context.TODO(), "cdi-operator", metav1.DeleteOptions{})
+		Expect(err).ToNot(HaveOccurred())
+	})
 })
 
+func infraDeploymentAvailable(f *framework.Framework, cr *cdiv1.CDI) bool {
+	cdi, _ := f.CdiClient.CdiV1beta1().CDIs().Get(context.TODO(), cr.Name, metav1.GetOptions{})
+	for _, c := range cdi.Status.Conditions {
```

Comment:
See https://github.com/kubevirt/containerized-data-importer/blob/master/pkg/operator/controller/controller_test.go#L158 for an example on code that does the same. import conditions and use IsStatusConditionTrue